### ISSUE TITLE
I don't manage to install jupyter in RDT build

### ISSUE THREAD

**Issue by 12rambau** [readthedocs/readthedocs.org#10223]
## Details

* Read the Docs project URL: https://readthedocs.org/projects/ipyvuetify
* Build URL (if applicable): https://readthedocs.org/projects/ipyvuetify/builds/20010475/
* Read the Docs username (if applicable): https://readthedocs.org/profiles/borntobealive/

## Expected Result

I would like to improve the documentation of the ipyvuetify lib and specifically build the API documentation. To do so I need to generate the widgets as there are not included in the files but generated upon installation. I need to install jupyter before installing the ipyvuetify lib to trigger the installation process.

```yaml
# Read the Docs configuration file
# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details

# Required
version: 2

# build with latest available ubuntu version
build:
  os: ubuntu-22.04
  tools:
    python: "3.10"
    nodejs: "14"
  jobs:
    pre_install:
      - python -m pip install jupyter

# Build documentation in the docs/ directory with Sphinx
sphinx:
  configuration: docs/conf.py

# Optionally set the version of Python and requirements required to build your docs
python:
  install:
    - method: pip
      path: .
      extra_requirements:
        - doc
```

## Actual Result

It works fine on my local computer but in RDT I repeatidly face the same issue (in https://github.com/widgetti/ipyvuetify/pull/240). 

```
Traceback (most recent call last):
        File "/home/docs/checkouts/readthedocs.org/user_builds/ipyvuetify/envs/240/bin/jupyter", line 5, in <module>
          from jupyter_core.command import main
      ModuleNotFoundError: No module named 'jupyter_core'
```

looking on the web this package is included in `jupyter` that (as shown in previous file) is pre-installed. 

do you have any idea why the lib is missing ?

**Comment by benjaoming** [comment]
@12rambau can you share a build URL?

**Comment by 12rambau** [comment]
copy paste error (the link leads to the build)

**Comment by 12rambau** [comment]
I did some further investigations and added the following small script in the pre-build command:

```
printf "import jupyter_core\nprint(jupyter_core.__version__)" | python
```

The build https://readthedocs.org/projects/ipyvuetify/builds/20092097/ shows that even when manually forcing jupyter_core installation, it's not installed. It's weird because I'm using the pre-build functionality for other packages and it works as expected. What makes Jupyter different ? Is ther anything from internal log reports that could help understand ?

**Comment by benjaoming** [comment]
Could this issue be helpful? => https://github.com/jupyter/jupyter_core/issues/295

**Comment by humitos** [comment]
I'm closing this issue because we haven't had any response in a month.

**Comment by 12rambau** [comment]
the linked error is still opened from jupyter side, I didn't saw anything there that can help me find a solution. The problem here is that I cannot reproduce the error locally even in fresh env and I have few knowledge on how things work on your side. So no their was no input from my side because I have nothing to add (the 14 last commit of this [PR](https://github.com/widgetti/ipyvuetify/pull/240) are trying to tune the RDT env without any effect). As this issue was not receiving any updates, I considered it was low priority from your side and opened a PR in ipyvuetify to refresh the building strategy and avoid installing jupyter alltogether: https://github.com/widgetti/ipyvuetify/pull/259, it's still waiting for review as well. 

Long story short: I didn't forget and tried to find solutions where I can act. If I'm the only one facing this and you need to clean the issue tracker, be it but for me this problem is still very active

**Comment by benjaoming** [comment]
@12rambau I didn't succeed to reproduce the issue like this... so I'm definitely a bit confused about what's happening.

1) Pull and run a Read the Docs build image:

   ```
   docker run -it readthedocs/build:ubuntu-22.04 /bin/bash
   ```

2) Firstly, install the environment:

   ```
   asdf install python 3.10.8`
   ```

3) Activate the Python env:

   ```
   asdf global python 3.10.8`
   ```

4) Then, run the same steps as your build process, for instance

   ```
   python -m pip install jupyter jupyter-packaging jupyterlab jupyter_core`
   ```

5) Try running the command:

   ```
   $ printf "import jupyter_core\nprint(jupyter_core.__version__)" | python
   5.3.0
   ```

   As you can see, this worked.


* Can you try running the failing build again? And could you add a step showing that some other random package is installed, i.e. `printf "import nbconvert\nprint(nbconvert.__version__)" | python`?
* Can you try to move the example from `pre_install` to `build.commands`. See: https://docs.readthedocs.io/en/stable/build-customization.html#build-commands-examples

**Comment by 12rambau** [comment]
sorry I'm stcked, the build you offer worked on the docker image but not on the website, I'm still waiting for reviews on my PR from ipyvuetify side and this problem should vanished as the install mechanism will be better. Sorry for keeping you on stale but I'm in the same situation.
